Replay cfgym

crccw opened the pull request:
<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/bnuacm/bnuoj-web-v3/25)

<!-- Reviewable:end -->

crccw committed:
add replay for cf gym

unofficial not included

crccw committed:
deal with .doc attachment when crawling cf gym EOM

51isoft commented:
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnuacm/bnuoj-web-v3/25)

---

<sup>**[ajax/admin_deal_replay.php, line 451 [r1]](https://reviewable.io:443/reviews/bnuacm/bnuoj-web-v3/25#-Jji-zgvYVlJqXGV6p_J)** ([raw file](https://github.com/bnuacm/bnuoj-web-v3/blob/3fdd2fab90d6c5ae945bde07ce46d4fcfa340e42/ajax/admin_deal_replay.php#L451)):</sup>
Remove spaces at the end

---

<sup>**[ajax/admin_deal_replay.php, line 455 [r1]](https://reviewable.io:443/reviews/bnuacm/bnuoj-web-v3/25#-Jji0hf7SFrMijeroGPH)** ([raw file](https://github.com/bnuacm/bnuoj-web-v3/blob/3fdd2fab90d6c5ae945bde07ce46d4fcfa340e42/ajax/admin_deal_replay.php#L455)):</sup>
Please add a helper php method like https://github.com/BNUACM/bnuoj-web-v3/blob/develop/ajax/openjudge_standing_merge.php

---

<!-- Sent from Reviewable.io -->

crccw committed:
fix reset function in problemlist EOM

crccw committed:
Alter replay_crawl_* to adapt to new problem adding list

Now instead of vpid, problems are passed down as a 'prob' array.

Be noted that JSON_FORCE_OBJECT for json_encode is required, jquery.populate cannot handle arrays

crccw committed:
return pnum for all replay_crawl_* EOM

crccw committed:
Use API instead when crawling CF Gym standing, and adding CF Gym to auto crawl

hhh.. Unofficial participants are included now, because ghost participants are treated as VIRTUAL on CF.

Anyway distinguish can also be made if necessary...

crccw committed:
remove spaces...EOM

crccw commented:
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnuacm/bnuoj-web-v3/25)

---

<sup>**[ajax/admin_deal_replay.php, line 451 [r1]](https://reviewable.io:443/reviews/bnuacm/bnuoj-web-v3/25#-Jji-zgvYVlJqXGV6p_J)** ([raw file](https://github.com/bnuacm/bnuoj-web-v3/blob/3fdd2fab90d6c5ae945bde07ce46d4fcfa340e42/ajax/admin_deal_replay.php#L451)):</sup>
Done.

---

<sup>**[ajax/admin_deal_replay.php, line 455 [r1]](https://reviewable.io:443/reviews/bnuacm/bnuoj-web-v3/25#-Jji0hf7SFrMijeroGPH)** ([raw file](https://github.com/bnuacm/bnuoj-web-v3/blob/3fdd2fab90d6c5ae945bde07ce46d4fcfa340e42/ajax/admin_deal_replay.php#L455)):</sup>
We now use API.

---

<!-- Sent from Reviewable.io -->

51isoft commented:
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bnuacm/bnuoj-web-v3/25)

---

<!-- Sent from Reviewable.io -->
